module/gui/MainWindow.py

```diff
@@ -16,54 +16,88 @@
+
 from PyQt4.QtCore import *
 from PyQt4.QtGui import *
 
-from os.path import join
+from os.path import join, dirname
+from datetime import datetime
 
 from module.gui.PackageDock import *
 from module.gui.LinkDock import *
-from module.gui.CaptchaDock import CaptchaDock
+from module.gui.CaptchaDialog import CaptchaDialog
 from module.gui.SettingsWidget import SettingsWidget
-
-from module.gui.Collector import CollectorView, Package, Link
+from module.gui.Collector import CollectorView
 from module.gui.Queue import QueueView
 from module.gui.Overview import OverviewView
 from module.gui.Accounts import AccountView
 from module.gui.AccountEdit import AccountEdit
-
-from module.remote.thriftbackend.ThriftClient import AccountInfo
+from module.gui.Tools import whatsThisFormat
 
 class MainWindow(QMainWindow):
-    def __init__(self, connector):
+    
+    def time_msec(self):
+        return int((datetime.utcnow() - datetime(1970, 1, 1)).total_seconds() * 1000 - self.time_msec_init)
+    
+    def __init__(self, corePermissions, appIconSet, connector):
         """
             set up main window
         """
+        self.time_msec_init = int((datetime.utcnow() - datetime(1970, 1, 1)).total_seconds() * 1000)
         QMainWindow.__init__(self)
+        self.setEnabled(False)
+        self.log = logging.getLogger("guilog")
+        self.corePermissions = corePermissions
+        self.appIconSet = appIconSet
+        self.connector = connector
+        self.lastAddContainerDir = unicode("")
+        
         #window stuff
+        self.setWindowFlags(self.windowFlags() | Qt.WindowContextHelpButtonHint)
         self.setWindowTitle(_("pyLoad Client"))
-        self.setWindowIcon(QIcon(join(pypath, "icons","logo.png")))
-        self.resize(1000,600)
+        self.setWindowIcon(QIcon(join(pypath, "icons", "logo.png")))
+        self.resize(100 ,100)
+        self.move(200 ,200)
+        self.initEventHooks()
         
         #layout version
         self.version = 3
         
         #init docks
-        self.newPackDock = NewPackageDock()
+        self.setDockOptions(QMainWindow.AnimatedDocks | QMainWindow.AllowTabbedDocks | QMainWindow.ForceTabbedDocks)
+        self.setTabPosition(Qt.RightDockWidgetArea, QTabWidget.North)
+        self.newPackDock = NewPackageDock(self.appIconSet)
         self.addDockWidget(Qt.RightDockWidgetArea, self.newPackDock)
         self.connect(self.newPackDock, SIGNAL("done"), self.slotAddPackage)
-        self.captchaDock = CaptchaDock()
-        self.addDockWidget(Qt.BottomDockWidgetArea, self.captchaDock)
-        self.newLinkDock = NewLinkDock()
+        self.connect(self.newPackDock, SIGNAL("parseUri"), self.slotParseUri)
+        self.newLinkDock = NewLinkDock(self.appIconSet)
         self.addDockWidget(Qt.RightDockWidgetArea, self.newLinkDock)
         self.connect(self.newLinkDock, SIGNAL("done"), self.slotAddLinksToPackage)
+        self.connect(self.newLinkDock, SIGNAL("parseUri"), self.slotParseUri)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['SIGNAL' may be undefined, or defined from star imports: PyQt4.QtCore, PyQt4.QtGui, module.gui.LinkDock, module.gui.PackageDock](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)